app.py

```diff
@@ -0,0 +1,138 @@
+    # put the records to kinesis
+    kinesis_client.put_records(
+        Records=kinesisRecords,
+        StreamName=kinesis_stream_name
+    )
+
+
+def decode_record(record):
+    try:
+        logger.info("Decoded event with eventId {}".format(record['eventID']))
+        return base64.b64decode(record["kinesis"]["data"]).decode('utf-8')
+    except TypeError as ex:
+        logger.error("Type Error: {}".format(ex))
+
+
+def filter_event(record, filters):
+    """
+    A Generic method that removes the provided parameter from the given json record.
+    :param record: Kinesis record in json representation
+    :param filters: parameter in events that are required to be removed
+    :return: filtered json record
+    """
+    try:
+        for filter in filters:
+            # splitting with '.' as we can have nested parameters that needs to be filtered, like body.type
+            values = filter.split('.')
+            if len(values) > 1:
+                expr = 'del record'
+                for element_name in values:
+                    expr = expr + '[\'' + element_name + '\']'
+                code = compile(expr, 'delete_record', 'exec')
+                eval(code)
+            else:
+                # if only one filter parameter present then just fetch that and delete it
+                del record[values[0]]
+        return record
+    except TypeError as ex:
+        logger.error("Error: {}".format(ex))
+
+
+"""
+This method filter the published records in CMSEvent Kinesis Queue.
+
+:param
+-event: The event stream
+"""
+
+
+@app.lambda_function()
+def execute(event, context):
+    # parameters to filter the stream, by default empty
+    to_filter_parameters = os.environ.get('TO_FILTER_PARAMETERS', '')
+    succeeded_record_count = 0
+    failed_record_count = 0
+    skipped_record_count = 0
+    # create a client with kinesis
+    kinesis = create_client('kinesis')
+
+    # if records are not present exit
+    if "Records" not in event:
+        logger.error("Not able to get records from arc cms event stream. Terminating the function")
+        sys.exit(1)
+
+    logger.info('Total events received: {}'.format(len(event['Records'])))
+    for record in event['Records']:
+        cms_event_json = decode_record(record)
+        try:
+            logger.info('Transforming the decoded event with partitionKey {} '
+                        'to json'.format(record["kinesis"]['partitionKey']))
+            event_data = json.loads(cms_event_json)
+            event_published_status = event_data['published']
+            logger.debug("event received with status PUBLISHED = {}. Event will be filtered is published=false"
+                         .format(event_published_status))
+            if event_published_status == bool('true'):
+                logger.debug("filtering the event by omitting parameter/'s' - {}".format(to_filter_parameters))
+                filtered_event = filter_event(event_data, to_filter_parameters.split(","))
```

Comment:
I think filter_event here does not make sense, in the method we are removing data elements so we should name the function appropriately, also documentation can be added with example about what the filters mean here